File: logipack-hub/hub-web/src/lib/server/hubApi/dto/employeeOffices.ts
import type { EmployeeDto } from "./employees";
import type { OfficeDto } from "./offices";

export type EmployeeOfficesResponseDto = {
	employee_id: string;

	offices: OfficeDto[];

	office_ids: string[];

	employee?: EmployeeDto | null;
	assigned_offices?: OfficeDto[] | null;
};

export type AssignEmployeeOfficeRequestDto = {
	office_id: string;
};

File: logipack-hub/hub-api/src/dto/employee_offices.rs
use serde::{Deserialize, Serialize};

use crate::dto::{employees::EmployeeDto, offices::OfficeDto};

#[derive(Debug, Serialize, Deserialize)]
pub struct AssignOfficeRequest {
    pub office_id: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ListEmployeeOfficesResponse {
    pub employee_id: String,
    pub offices: Vec<OfficeDto>,
    pub office_ids: Vec<String>,
    pub employee: Option<EmployeeDto>,
    pub assigned_offices: Option<Vec<OfficeDto>>,
}
